Add change password functionality in Auth module

PreranaMore24 opened the pull request:
Implement a new PATCH endpoint for changing user passwords in the AuthController. The changePassword method in AuthService validates the current password and updates it to a new hashed password. This enhances user account security by allowing password updates.

PreranaMore24 committed:
Add change password functionality in Auth module

Implement a new PATCH endpoint for changing user passwords in the AuthController. The changePassword method in AuthService validates the current password and updates it to a new hashed password. This enhances user account security by allowing password updates.

vedantkirve commented on src/auth/auth.service.ts:
```diff
@@ -204,6 +205,43 @@ export class AuthService {
+
+    const hashedPassword = await this.hashPassword(newPassword);
+    await this.prisma.user.update({
+      where: { id: userId },
```
can not call user realted query in auth module .
Already Ig update User exists user that function

PreranaMore24 replied:
Existing UserService function `updateUserProfile`  only update general profile fields like firstname lastname 
Should create a seperate dunction to update password?

vedantkirve commented on src/auth/auth.controller.ts:
```diff
@@ -57,4 +69,11 @@ export class AuthController {
+  @UsePipes(new ZodValidationPipe(ChangePasswordSchema))
+  async changePassword(@Req() req, @Body() dto: ChangePasswordDto) {
+    const userId = req.user.userId;
+    return this.authService.changePassword(userId, dto);
```
try catch needed here catch the error in controller take reference from other controller

PreranaMore24 committed:
Refactor change password logic and add error handling

Enhance the change password functionality in AuthController by adding error handling for user ID retrieval and service call. Introduce a new updateUserPassword method in UserService to encapsulate password update logic with proper exception handling for user existence and update errors. This improves robustness and user feedback during password changes.
